services/scabbard/src/cli/transaction.rs

```diff
@@ -0,0 +1,402 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Write;
+use std::iter::FromIterator;
+use std::time::Instant;
+
+use protobuf::{self, Message};
+use sabre_sdk::{
+    protocol::{
+        payload::{Action, SabrePayload},
+        ADMINISTRATORS_SETTING_ADDRESS,
+    },
+    protos::IntoBytes,
+};
+use sawtooth_sdk::messages::{
+    batch::{Batch, BatchHeader, BatchList},
+    transaction::{Transaction, TransactionHeader},
+};
+use splinter::signing::{hash::HashSigner, Signer};
+
+use super::error::CliError;
+
+/// Returns a Transaction for the given Payload and Signer
+///
+/// # Arguments
+///
+/// * `payload` - a fully populated identity payload
+/// * `signer` - the signer to be used to sign the transaction
+///
+/// # Errors
+///
+/// An error is returned if it occurs during serialization of the provided payload or internally
+/// created `TransactionHeader`, or if a signing error occurs.
+pub fn create_transaction(
```

Comment:
We should really put into_transaction_builder() into the Sabre SDK (would essentially be this code minus the signer).

Reply:
I agree with this and all the other comments about using Transact/Sabre SDK to do all these things for us. In the interest of getting this merged and the functionality available, would it be fine to open stories to update Sabre and Transact to support the desired behavior and make the corresponding changes here?

Reply:
Created a story to implement this capability